File: produtos/src/main/java/br/com/bota/service/ProdutoService.java
package br.com.bota.service;

import br.com.bota.entity.Produto;

import java.util.List;
import java.util.UUID;

public interface ProdutoService {
    Produto salvar(Produto product);
    void baixarEstoque(UUID id, Integer quantidade);
    Produto procurarPorId(UUID id);
    Boolean verificarQuantidade(UUID id);
    List<Produto> procurar();
}

File: loja-kotlin/kprodutos/src/main/kotlin/br/com/bota/service/ProdutoService.kt
package br.com.bota.service

import br.com.bota.entity.Produto
import java.util.*

interface ProdutoService {
    fun salvar( produto: Produto) : Produto
    fun baixarEstoque(id: UUID, quantidade:Int)
    fun procurarPorId(id: UUID ): Produto
    fun verificarQuantidade(id: UUID ): Boolean
    fun procurar() : List<Produto>
}